pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -597,6 +597,17 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
 		appInstance.FixedResources.VncPasswd = cfgApp.Fixedresources.VncPasswd
 		appInstance.FixedResources.DisableLogs = cfgApp.Fixedresources.DisableLogs
 		appInstance.MetaDataType = types.MetaDataType(cfgApp.MetaDataType)
+		// Calculate the moment when the application should start, taking into account the configured delay
+		startTime := getconfigCtx.zedagentCtx.delayBaseTime.Add(time.Duration(cfgApp.StartDelayInSeconds) * time.Second)
+		appInstance.StartTime = startTime
```

Comment:
With this code moved to zedmanager then AppInstanceConfig can just have a StartDelayInSeconds field copied just like the other fields, with the logic to compute the earliest start time belonging in zedmanager,

Reply:
The same. I can move it to zedmanager and let's see.

Reply:
FWIW we want keep zedagent as simple as possible by mapping between the protobuf and golang structs. There are cases where it does a lot more e.g., to handle parse errors, non-trivial mappings with ordering issues, etc.

Reply:
Thanks for this clarification. It will help in the future as well)

Reply:
Moved the calculation into the zedmanger.